01_fib/ailyntang/fib_test.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+)
+
+func TestUnexpectedInputs(t *testing.T) {
+	errorString := strconv.Quote("fib.go only handles integers between 1 to 92 inclusive\n")
+
+	tests := []struct {
+		input    int
+		expected string
+	}{
+		{0, errorString},
+		{93, errorString},
+		{183, errorString},
+		{-5, errorString},
+		{-205, errorString},
+	}
+
+	for _, test := range tests {
+		var buf bytes.Buffer
+		out = &buf
+
+		fib(test.input)
+
+		expected := test.expected
+		actual := strconv.Quote(buf.String())
+
+		if expected != actual {
+			t.Errorf("Unexpected response from fib(%v)", test.input)
+			t.Errorf("expected: %v\nactual: %v", expected, actual)
+		}
+	}
+}
+
+func TestGenerateFibSequence(t *testing.T) {
+	tests := []struct {
+		input    int
+		expected []int64
+	}{
+		{1, []int64{1}},
+		{7, []int64{1, 1, 2, 3, 5, 8, 13}},
+		{10, []int64{1, 1, 2, 3, 5, 8, 13, 21, 34, 55}},
+	}
+
+	for _, test := range tests {
+		actual := genFib(test.input)
+		expected := test.expected
+		errorMessage := "Incorrect Fibonnaci sequence\nexpected: %v\nactual: %v"
+
+		if (expected == nil) != (actual == nil) {
```

Comment:
This comparison is not really needed as far as I can see, and it becomes `if actual == nil` as `expected` is always not `nil`. But that comparison is not needed either since the comparisons afterwards will fail. If you are concerned about trying to take the length of a nil slice or range over it, you don't need to be. A `nil` slice is valid. See https://tour.golang.org/moretypes/12 for a simple example.

So I would just remove this `if` altogether and just rely on the length comparison and contents comparison.

Reply:
Thanks for the link and explanation. You're right, I was concerned that taking a length of a nil slice would produce an error. Good to know that's fine. I'll remove this `if` statement.